subproviders/wallet.js

```diff
@@ -13,7 +13,7 @@ function WalletSubprovider (wallet, opts) {
   }
 
   opts.getPrivateKey = function (address, cb) {
-    if (address !== wallet.getAddressString()) {
+    if (address.toLowerCase() !== wallet.getAddressString()) {
```

Comment:
if we are sure `wallet.getAddressString()` always returns a string would it make sense to call a `.toLowerCase()` on its result also?

Reply:
It always returns a lowercase string. There is `getChecksummedAddressString()` for the checksummed one.

Reply:
I would add a check to see whether address has been provided, otherwise calling `toLowerCase()` on anything other than a string will throw.

Would also store the result of `wallet.getAddressString()` in a variable.